source/ui/acquaman/AMScanConfigurationViewHolder3.cpp

```diff
@@ -133,15 +138,13 @@ AMAction3* AMScanConfigurationViewHolder3::createMultipleScans()
 	if (view_){
 
 		const AMScanConfiguration *config = view_->configuration();
+
 		if (config) {
-			int iteration = iterationsBox_->value();
-			if (iteration > 1) {
-				AMLoopAction3 *loopAction = new AMLoopAction3(new AMLoopActionInfo3(iterationsBox_->value(), config->name(), config->description()));
-				loopAction->addSubAction(createScan());
-				return loopAction;
-			} else {
-				return createScan();
-			}
+
+			AMLoopAction3 *loopAction = new AMLoopAction3(new AMLoopActionInfo3(iterationsBox_->value(), config->name(), config->description()));
+			loopAction->addSubAction(createScan());
+
+			return loopAction;
```

Comment:
Correct.  If you instantiate the view holder such that it you want it to create loops (`enableLoopAction = true` in the constructor) it will always create loops regardless of the number of scans you select to start.  If `enableLoopAction` is false then it will simply create the single scan like it always has.